server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -394,6 +402,23 @@ private static long jitter() {
         * TabletServer.TIME_BETWEEN_LOCATOR_CACHE_CLEARS);
   }
 
+  public Optional<Semaphore> getSemaphore() {
+    int configuredWriteThreadsMax =
+        getServerConfig().getConfiguration().getCount(Property.TSERV_WRITE_THREADS_MAX);
+    if (configuredWriteThreadsMax == MAX_WRITE_THREADS_DEFAULT) {
+      return Optional.empty();
```

Comment:
This basically returns a no-op. If the configured is equal to the default, do nothing.

Reply:
Yeah, but if the default is changed... then the semantics change. That only works because the default is `0`. That's a very roundabout way of comparing to `0`. We shouldn't care if it's equal to the default... we should care if it's equal to `0`, regardless of what the default is.

Reply:
It would be very easy for someone to change the default value without changing the underlying code. I was thinking that new method would be a good way to guard against future coding errors. I won't be around forever :smile: